### ISSUE TITLE
Let a caller ask for a layout at construction, and let a backend refuse

### ISSUE THREAD

**Issue by xupremix** [xupremix/incin#115]
## What

Two places pin the memory order to row-major and cannot express anything else.

`crates/incin-backends/src/target/place.rs`:

```rust
pub type TargetTensor<T, S, K> = Tensor<S, TargetBackend<T>, K, NoGrad>;
```

Six parameters, five named — the layout slot takes its default, so nothing the target API returns can carry a proof. And every data constructor funnels through `allocate_row_major` in `target/ext.rs`, which hardcodes the answer in its name.

The core constructors are already generic over the layout, bounded on `FreshDense<S>`, so `let t: Dense<s![3, 4], B> = Tensor::zeros(())?` works today. But `FreshDense` has exactly two implementors — `Dyn` and `RowMajor` — and it is sealed on the premise that *a fresh allocation genuinely is both*. That premise is the whole design, and it stops holding the moment someone wants a layout a fresh allocation is not.

## Proposal

Generalise the sealed trait so it carries the strides to allocate with:

```rust
pub trait FreshLayout<S: Shape>: LayoutOf<S> {
    /// The strides an allocation must use for `Self` to describe it.
    fn strides(dims: &[usize]) -> StrideBuf;
}
```

The constructor calls `L::strides(dims)`, hands them to the backend, and returns a tensor claiming `L`. The claim is then honest **by construction**: the value that produced the strides is the one named in the type, so there is no gap between what was asked for and what was allocated.

That also dissolves the reason for the seal. A downstream author writing `ChannelsLast::strides` incorrectly is lying about their own type, which is the trust `Layout::STATIC_STRIDES` already extends. The seal closed a *different* hazard — an unbounded constructor stamping a layout onto an allocation that had nothing to do with it — and that is impossible once the layout chose the allocation.

Spelling stays type-directed, matching the existing `Dense<S, B>` return-position pattern rather than a `memory_format`-style argument, which would introduce a value that has to agree with a type.

## The constraint that decides whether this is sound

**Backends must be able to refuse.** CPU and CUDA allocate contiguous today. A `ChannelsLast` request must either be honoured or rejected through the existing `*_layouts` capability rows — never silently allocated row-major and returned wearing a channels-last type.

A creation API that cannot refuse is a minting press with extra steps, which is the exact hazard the seal was introduced to close.

## Why this needs a second layout to exist

Three of the twelve worked cases in the design note are untestable today:

- **`reshape_view` must stop compiling on a non-contiguous layout.** `Contiguous` has never had to *exclude* anything — both implementors satisfy it — so that bound has never been tested. A second layout is what turns it from decoration into a check.
- **A pointwise result must not carry channels-last onto a row-major buffer.** This is the case that shows "state, don't carry" was right rather than merely defensible.
- **`Dropout`'s bound is currently too tight to accept such a layout at all**, since `FreshDense` means "a dense buffer satisfies this".

So `ChannelsLast` and `FreshLayout` should land in the same change, with a conformance test asserting both strides and values. A creation API designed against `RowMajor` alone is fitted to the one case that needed no API.

## Prior art

Surveyed in the design note. The headline, worth stating in the book: **incin's `Layout` is PyTorch's `memory_format`, not PyTorch's `layout`** — PyTorch's `layout=` is the storage scheme (`strided` vs `sparse_coo`), and `memory_format=` is the stride order. PyTorch's `*_like` default is `preserve_format`, i.e. carry, which is the contract removed in the density work; its propagation through ops is precedence heuristics rather than a rule, and the tag being a runtime fact is why `.view()` still raises at runtime.

JAX/XLA treats layout as a compiler concern with a `jit`-boundary escape hatch. TensorFlow puts it on the operation (`data_format=`) rather than the value. CuTe is where `LayoutOf` came from.

## Open questions

- Does `Contiguous` stay a marker or become `Contiguous<Order>`? A blocked layout is contiguous in a sense `reshape_view` still cannot use.
- Should `transpose_view` restate the layout the way `into_shape` now does? A transposed `RowMajor<[H, W]>` is a real layout of `[W, H]`, just not `RowMajor`.
- Does the capability table gate creation by `LayoutClass`, or does creation need its own row? `Strided` is too coarse: a backend may allocate channels-last and still refuse an arbitrary stride.

## Reference

`docs/plan/research/0.2.0/layout-at-construction.md` — full design note with twelve worked cases.

**Comment by xupremix** [comment]
Substantially implemented since this was filed. Status against `master`:

**Done.**
- `FreshLayout<S>` exists, carrying `fn strides(dims) -> StrideBuf`, unsealed for the reason given here — nothing about it can be minted, because the claim cannot outrun an allocation the claim itself chose.
- The target API expresses a layout: `TargetTensorIn`, `tensor_in`, `zeros_in`, and the `allocate_in` hook. The layout is the *first* type parameter, so the common call needs no turbofish — the annotation naming the proof chooses the layout.
- Backends can refuse: `HostInterop::from_bytes_strided` is defaulted to a capability refusal, so the contract stayed additive and a backend that has not opted in cannot be asked for a non-dense allocation.
- `ChannelsLast<S>` exists **and is constructible**. CPU opts in; `tensor_in` permutes host data into the layout's order via `shapes::scatter_positions`.
- `Tensor::into_layout::<L>()`, the general form of `into_row_major`.

**Case 8 was the one to get right, and it needed three attempts.** Both `to_vec1` and `to_bytes` read *by the strides* — `to_bytes` is canonical on purpose, or a saved tensor could not be reloaded — so the physical buffer is not observable through the public API at all. What makes the round-trip discriminating is arithmetic: a row-major walk of `[1, 2, 2, 2]` visits offsets 0, 2, 4, 6, 1, 3, 5, 7 under strides `[8, 1, 4, 2]`, so an unpermuted buffer reads back as `[1, 3, 5, 7, 2, 4, 6, 8]`. That is now an explicit negative control in the test.

**Still open.**
- `zeros_in` refuses `ChannelsLast`: a creation kernel writes dense and nothing permutes afterwards. The upload path has host data to reorder; the creation path does not. Honest asymmetry rather than an oversight, but it is a real limitation.
- Only CPU has opted into `from_bytes_strided`.
- The three open questions at the bottom of this issue are untouched: whether `Contiguous` becomes `Contiguous<Order>`, whether `transpose_view` should restate its layout, and whether creation needs its own capability row.
